sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SymmetricHashJoinStateManager.scala

```diff
@@ -185,6 +187,57 @@ class SymmetricHashJoinStateManager(
     }
   }
 
+  /**
+   * Perform a full scan to provide all available data.
+   *
+   * This produces an iterator over the (key, value, match) tuples. Callers are expected
+   * to consume fully to clean up underlying iterators correctly.
```

Comment:
why is this the case? Don't you want to support limit pushdown?

Reply:
Let's file a ticket to address altogether.